common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -701,10 +701,32 @@ def get_youtube_link(video_id):
             'default_value': [get_youtube_link(youtube_id_1_0['default_value'])]
         })
 
-        youtube_id_1_0_value = get_youtube_link(youtube_id_1_0['value'])
+        source_url = self.create_youtube_url(youtube_id_1_0['value'])
```

Comment:
@muhammad-ammar  `get_youtube_link` fetches YT val encoding then invokes self.create_youtube_url with that val encoding if it exists otherwise simply creates youtube url with passed YT id.

I'm doing only otherwise part here and as far as val YT encoding is concerned, it'll get into the source url if it exists in val – i.e. in later conditionals.